src/schema/mod.rs

```diff
@@ -45,16 +49,79 @@ pub struct Assertion {
     pub version: String,
     #[serde(rename = "Issuer")]
     pub issuer: Issuer,
-    #[serde(rename = "ds:Signature")]
+    #[serde(rename = "Signature")]
     pub signature: Option<Signature>,
     #[serde(rename = "Subject")]
     pub subject: Option<Subject>,
     #[serde(rename = "Conditions")]
     pub conditions: Option<Conditions>,
     #[serde(rename = "AuthnStatement")]
-    pub authn_statements: Vec<AuthnStatement>,
+    pub authn_statements: Option<Vec<AuthnStatement>>,
     #[serde(rename = "AttributeStatement")]
-    pub attribute_statements: Vec<AttributeStatement>,
+    pub attribute_statements: Option<Vec<AttributeStatement>>,
+}
+
+impl Assertion {
+    fn name() -> &'static str {
+        "saml2:Assertion"
+    }
+
+    fn schema() -> Vec<(&'static str, &'static str)> {
+        vec![
+            ("xmlns:saml2", "urn:oasis:names:tc:SAML:2.0:assertion"),
+            ("xmlns:xsd", "http://www.w3.org/2001/XMLSchema"),
+        ]
+    }
```

Comment:
```suggestion
    fn schema() -> &'static [(&'static str, &'static str)] {
        &[
            ("xmlns:saml2", "urn:oasis:names:tc:SAML:2.0:assertion"),
            ("xmlns:xsd", "http://www.w3.org/2001/XMLSchema"),
        ]
    }
```
Doesn't seem to be a need to allocate for this